### PR TITLE
Update to Quarkus 2.14.1.Final

### LINKED ISSUE #15329: Update to Quarkus 2.14.1.Final
### Description

The Quarkus team released 2.14.0.Final yesterday with some bug fixes, we should be able to update.

Related issues: #14925

### PR DESCRIPTION
**Pull request by abstractj**
Closes #15329

Supersedes #15531

### CODE REVIEW

**Comment by abstractj** [comment]
FYI @vmuzikar

**Comment by abstractj** [comment]
@pedroigor executing our update script, and setting the Quarkus release to 14.0.1.Final will downgrade most of the Infinispan dependencies that the storage team needs. I still need to check why.

Also, it will introduce a new unused property `javax.annotation-api.version`.

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@abstractj Thanks for taking over the PR.

I believe the [original comment](https://github.com/keycloak/keycloak/pull/15531#pullrequestreview-1183349546) from @pedroigor still applies. Looking at [Quarkus POM](https://github.com/quarkusio/quarkus/blob/2.14.1.Final/bom/application/pom.xml#L22), I believe we should upgrade RESTEasy at least.

File: pom.xml [hunk 1]
```diff
@@ -83,6 +83,7 @@
         <hibernate.c3p0.version>5.6.10.Final</hibernate.c3p0.version>
         <infinispan.version>14.0.2.Final</infinispan.version>
         <infinispan.protostream.processor.version>4.5.0.Final</infinispan.protostream.processor.version>
+        <javax.annotation-api.version>1.3.2</javax.annotation-api.version>
```
Comment (abstractj) [inline pom.xml:86]: @vmuzikar this is the unused property which I mentioned to you.

Comment (vmuzikar) [reply pom.xml:86]: Yes, this property was removed [here](https://github.com/keycloak/keycloak/commit/5e891951f5edadb5ad7a595e4e34203e175668c5#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8L86).

I'll try to investigate. I think we need to update the `set-quarkus-version.sh` to reflect latest changes.

**Pull request closed without merging**